cli/bptest/cmd.go

```diff
@@ -145,3 +146,32 @@ var initCmd = &cobra.Command{
 		return initTest(initTestName)
 	},
 }
+
+var lintCmd = &cobra.Command{
+	Use:   "lint",
+	Short: "Lint Blueprint connect references and versions",
+	Long:  "Lint Blueprint connect references and versions",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Run the existing lint tests
+		fmt.Println("Running lint tests...")
+
+		// Walk the directory to find all metadata.yaml files
+		err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() || info.Name() != "metadata.yaml" {
+				return nil
+			}
+			return validateMetadataFile(path)
```

Comment:
Could we define a common metadata lint interface and make each rule a concrete implementation? We will then register all the lint rules and invoking this command would run through them.

The advantage is this allows us to then easily do targeted linting, exclusions etc and writing new lint rules. The base  runner holds common code for parsing metadata, hcl etc. As an example, this is tflint's interface if you were to implement a custom rule - https://github.com/terraform-linters/tflint-plugin-sdk/blob/8c324098348dc54377b4b0f0bcf6c4e3be078c2a/tflint/interface.go#L316

Reply:
Done